webapp/src/js/services/db.js

```diff
@@ -55,13 +57,27 @@ angular.module('inboxServices').factory('DB',
       return clone;
     };
 
+    const emitEventOnWrite = function(pouchDb) {
+      const pouchDbWriteOperations = ['putAttachment', 'put', 'bulkDocs'];
+      pouchDbWriteOperations.forEach(function(operation) {
+        pouchDb[operation] = _.wrap(pouchDb[operation], function(original) {
+          const args = Array.from(arguments).splice(1);
+          $rootScope.$emit('dbWriteEvent', { operation, args });
+          return original.apply(pouchDb, args);
+        });
+      });
+    };
```

Comment:
I'm uncomfortable with wrapping - it's easy to miss some stubs (eg: `post`), it's harder to read, and it might conflict with other libraries we may use (angular-pouch, worker-pouch, etc). Can we instead use the `Changes` service? It'll fire more often because it'll pick up replications too but I think the `isSyncInProgress` check will ignore them so it should be ok.

This also means we can drop the `$rootScope` dependency and event.

Reply:
Ya.... I really missed the boat here.